File: salesforge/backend/app/schemas/company.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CompanyCreate(BaseModel):
    name: str
    website: str | None = None
    industry: str | None = None
    phone: str | None = None
    address: str | None = None
    notes: str | None = None


class CompanyUpdate(BaseModel):
    name: str | None = None
    website: str | None = None
    industry: str | None = None
    phone: str | None = None
    address: str | None = None
    notes: str | None = None


class CompanyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    website: str | None
    industry: str | None
    phone: str | None
    address: str | None
    notes: str | None
    owner_id: str
    created_at: datetime
    updated_at: datetime


class LinkedContactResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    first_name: str
    last_name: str
    email: str | None
    title: str | None


class LinkedDealResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    title: str
    stage: str
    value: float | None
    expected_close_date: object | None = None


class CompanyDetailResponse(CompanyResponse):
    contacts: list[LinkedContactResponse] = []
    deals: list[LinkedDealResponse] = []


class CompanyListResponse(BaseModel):
    items: list[CompanyResponse]
    total: int
    page: int
    page_size: int

File: salesforge/frontend/src/types/company.ts
export interface Company {
  id: string
  name: string
  website: string | null
  industry: string | null
  phone: string | null
  address: string | null
  notes: string | null
  owner_id: string
  created_at: string
  updated_at: string
}

export interface LinkedContact {
  id: string
  first_name: string
  last_name: string
  email: string | null
  title: string | null
}

export interface CompanyLinkedDeal {
  id: string
  title: string
  stage: string
  value: number | null
  expected_close_date: string | null
}

export interface CompanyDetail extends Company {
  contacts: LinkedContact[]
  deals: CompanyLinkedDeal[]
}

export interface CompanyListResponse {
  items: Company[]
  total: number
  page: number
  page_size: number
}

export interface CompanyCreate {
  name: string
  website?: string
  industry?: string
  phone?: string
  address?: string
  notes?: string
}

export type CompanyUpdate = Partial<CompanyCreate>
